x/mint/keeper/keeper_test.go

```diff
@@ -60,109 +60,130 @@ func TestKeeperTestSuite(t *testing.T) {
-			Weight:  sdk.NewDec(1),
+			name:         "coin is minted to the fee collector",
+			fee:          sdk.NewCoin("stake", sdk.NewInt(0)),
+			ratio:        sdk.NewDecWithPrec(2, 1),
+			expectedCoin: sdk.NewCoin("stake", sdk.NewInt(0)),
+		}, {
+			name:  "mint the 100K stake coin to the fee collector",
+			fee:   sdk.NewCoin("stake", sdk.NewInt(100000)),
+			ratio: sdk.NewDecWithPrec(2, 1),
+			fn: func() {
+				fee := sdk.NewCoin("stake", sdk.NewInt(100000))
+				fees := sdk.NewCoins(fee)
+				err := simapp.FundModuleAccount(suite.App.BankKeeper,
+					suite.Ctx,
+					authtypes.FeeCollectorName,
+					fees)
+				suite.NoError(err)
+			},
+			expectedCoin: sdk.NewCoin("stake", sdk.NewInt(100000).Quo(sdk.NewInt(5))),
 		},
 	}
-	suite.App.MintKeeper.SetParams(suite.Ctx, params)
-
-	// Create record
-	coins := sdk.Coins{sdk.NewInt64Coin("stake", 10000)}
-	suite.FundAcc(gaugeCreator, coins)
-	distrTo := lockuptypes.QueryCondition{
-		LockQueryType: lockuptypes.ByDuration,
-		Denom:         "lptoken",
-		Duration:      time.Second,
-	}
-
-	// mints coins so supply exists on chain
-	mintLPtokens := sdk.Coins{sdk.NewInt64Coin(distrTo.Denom, 200)}
-	suite.FundAcc(gaugeCreator, mintLPtokens)
-
-	gaugeId, err := suite.App.IncentivesKeeper.CreateGauge(suite.Ctx, true, gaugeCreator, coins, distrTo, time.Now(), 1)
-	suite.NoError(err)
-	err = suite.App.PoolIncentivesKeeper.UpdateDistrRecords(suite.Ctx, poolincentivestypes.DistrRecord{
-		GaugeId: gaugeId,
-		Weight:  sdk.NewInt(100),
-	})
-	suite.NoError(err)
 
-	// At this time, there is no distr record, so the asset should be allocated to the community pool.
-	mintCoin := sdk.NewCoin("stake", sdk.NewInt(100000))
-	mintCoins := sdk.Coins{mintCoin}
-	err = mintKeeper.MintCoins(suite.Ctx, mintCoins)
-	suite.NoError(err)
-	err = mintKeeper.DistributeMintedCoin(suite.Ctx, mintCoin)
-	suite.NoError(err)
+	for _, tc := range tests {
+		suite.Run(tc.name, func() {
+			// reset suite
+			suite.SetupTest()
+			coin := suite.App.MintKeeper.GetProportions(suite.Ctx, tc.fee, tc.ratio)
+			suite.Equal(tc.expectedCoin, coin)
 
-	feePool := suite.App.DistrKeeper.GetFeePool(suite.Ctx)
-	feeCollector := suite.App.AccountKeeper.GetModuleAddress(authtypes.FeeCollectorName)
-	suite.Equal(
-		mintCoin.Amount.ToDec().Mul(params.DistributionProportions.Staking).TruncateInt(),
-		suite.App.BankKeeper.GetAllBalances(suite.Ctx, feeCollector).AmountOf("stake"))
-	suite.Equal(
-		mintCoin.Amount.ToDec().Mul(params.DistributionProportions.CommunityPool),
-		feePool.CommunityPool.AmountOf("stake"))
-	suite.Equal(
-		mintCoin.Amount.ToDec().Mul(params.DistributionProportions.DeveloperRewards).TruncateInt(),
-		suite.App.BankKeeper.GetBalance(suite.Ctx, devRewardsReceiver, "stake").Amount)
+		})
+	}
+}
 
-	// Test for multiple dev reward addresses
-	params.WeightedDeveloperRewardsReceivers = []types.WeightedAddress{
+func (suite *KeeperTestSuite) TestDistrAssetToDeveloperRewardsAddrWhenNotEmpty() {
```

Comment:
What does `NotEmpty` mean in this context?

Reply:
It's distrRecord. I removed it in the new commit because it could be unintelligible.